python_modules/dagster/dagster/_utils/test/postgres_instance.py

```diff
@@ -181,17 +176,13 @@ def docker_service_up(
 
         conn_str = TestPostgresInstance.conn_string(**conn_args)
         wait_for_connection(conn_str, retry_limit=10, retry_wait=3)
-        yield conn_str
-
         try:
-            subprocess.check_output(
-                ["docker-compose", "-f", docker_compose_file, "stop", service_name]
-            )
-            subprocess.check_output(
-                ["docker-compose", "-f", docker_compose_file, "rm", "-f", service_name]
+            yield conn_str
+        finally:
+            subprocess.run(
+                ["docker-compose", "-f", docker_compose_file, "down", service_name],
+                check=False,
             )
-        except subprocess.CalledProcessError:
```

Comment:
I follow how this will help limit dangling containers. I don't think I follow how the other changes (when only the `docker_compose` fixture is being used) will help. Is the issue caused by a weird interaction between the two (using both the postgres test util and the `docker_compose` fixture simultaneously)?

Reply:
the changes here are
* moving from `check_` with swallowing the error to `run` with `check=False`
* putting the clean-up in the `finally` block so if we bubble out through this context manager via exception it still executes 
* swapping `stop` + `rm` to just `down` which does both operations in one go